### PR TITLE
Pin to Node Selection Behaivior

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose

The purpose of this PR is to address some existing issues found on the pin note to node behavior. 
To guarantee a user friendly experience when using the `pin to node` functionality we made sure that:
-  `Pin to node` is enabled only when 1 note and 1 node are selected.
- `Pin to node`  is disabled when trying to pin a note to a node that is already pinned to another note.
-  When selecting a pinned note & note, previous selection is overwritten, instead of being added to selection.
-  Tooltip added explaining how to pin to a node.

pinToNodeSelection2

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@Amoursol @QilongTang 

### FYIs

@SHKnudsen

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]
LGTM

File: src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs [hunk 1]
```diff
@@ -305,11 +305,23 @@ private bool CanPinToNode(object parameters)
...
+                return false;
+
+            var nodeToPin = nodeSelection.FirstOrDefault();
+
+            var nodeAlreadyPinned = WorkspaceViewModel.Notes
+                .Where(n => n.PinnedNode != null)
+                .Any(n => n.PinnedNode.NodeModel.GUID == nodeToPin.GUID);
 
-            if (nodeToPin == null || Model.PinnedNode != null)
+            if (nodeToPin == null || Model.PinnedNode != null || nodeAlreadyPinned)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs:324]: @martinstacey Can we add a unit test for this?

**Comment by Amoursol** [comment]
Looks great @martinstacey !

File: src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs (hunk 1 above)
Comment (martinstacey) [reply src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs:324]: Ok test added on this commit: 3a34af0

**Pull request merged**